tests/e2e/PreMutantFilter/filter/ExampleFilter.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace PreMutantFilter;
+
+use Infection\Plugins\MutantFilterPlugin;
+
+class ExampleFilter implements MutantFilterPlugin
+{
+    private $count;
+
+    private function __construct()
+    {
+        $this->count = 0;
+    }
+
+    public static function create()
+    {
+        return new static();
+    }
+
+    public function getMutantFilter(): ?callable
+    {
+        return function (string $pathToMutatedFile) {
```

Comment:
This has to be extended to include more than just path. For example, if one knows that running static analysis for a file takes a second, and running tests will take half a second, it would make sense to skip the analysis part testing the mutation anyways.